src/components/application_manager/src/application_state.cc

```diff
@@ -291,14 +288,7 @@ void ApplicationState::SetPostponedState(const WindowID window_id,
                         HmiState::StateID::STATE_ID_POSTPONED);
 
   sync_primitives::AutoLock auto_lock(hmi_states_map_lock_);
-  HmiStates& hmi_states = hmi_states_map_[window_id];
-  DCHECK_OR_RETURN_VOID(!hmi_states.empty());
-  HmiStatePtr front_state = hmi_states.front();
-  if (front_state->state_id() == HmiState::StateID::STATE_ID_POSTPONED) {
-    hmi_states.erase(hmi_states.begin());
-  }
-
-  hmi_states.insert(hmi_states.begin(), state);
+  postponed_states_map_[window_id] = state;
```

Comment:
Should postponed_states_map_ require its own lock instead of sharing the hmi_states_map_lock_?